### ISSUE TITLE
error: Please make sure you are using the server version of the FTB modpack!

### ISSUE THREAD

**Issue by ibootstrapper** [itzg/docker-minecraft-server#743]
I'm using the below docker-compose.yml file to run the server with RLCRaft though it exits with an error: "Please make sure you are using the server version of the FTB modpack!" any ideas what i'm doing wrong?
```
version: '3'
services:
  minecraft:
    ports:
      - "25565:25565"
    volumes:
            - "./data:/data"
    environment:
      MAX_MEMORY: 4G
      DIFFICULTY: hard
      VERSION: 1.12.2
      TYPE: CURSEFORGE
      CF_SERVER_MOD: modpacks/RLCraft+1.12.2+-+Beta+v2.8.2.zip
    image: itzg/minecraft-server
    restart: always
  rcon:
    image: itzg/rcon
    ports:
      - "4326:4326"
      - "4327:4327"
    volumes:
            - "./db:/opt/rcon-web-admin/db"
  ```

**Comment by itzg** [comment]
From the file name it looks like you may not be using the server version of the modpack. Try instead:

https://www.curseforge.com/minecraft/modpacks/rlcraft/files/2935323

**Comment by isamuelson** [comment]
I ended up changing to the following and it works though there are many time lags
```
version: '3'

services:
  rlcraft:
    ports:
      - "25565:25565"
    environment:
      EULA: "TRUE"
      MAX_MEMORY: 4G
      MAX_BUILD_HEIGHT: 200
      VIEW_DISTANCE: 10
      LEVEL_TYPE: LARGEBIOMES
      MAX_PLAYERS: 100
      CONSOLE: "false"
      DIFFICULTY: hard
      VERSION: 1.12.2
      TYPE: FORGE
      FORGEVERSION: 14.23.5.2838
      GENERIC_PACK: modpacks/RLCraft+Server+Pack+1.12.2+-+Beta+v2.8.2.zip
      GENERIC_PACK_STRIP_DIRS: "1"
      DISABLE_HEALTHCHECK: "true"
    image: itzg/minecraft-server
    restart: always
    volumes:
      - ./data:/data
      - ./modpacks:/modpacks

  rcon:
    image: itzg/rcon
    ports:
      - "4326:4326"
      - "4327:4327"
    volumes:
      - "./db:/opt/rcon-web-admin/db" #

```

**Comment by itzg** [comment]
Nice workaround! I now vaguely remember having to do something like that for RLcraft.

**Comment by itzg** [comment]
Can this issue be closed now? Lag will be caused by factors external to the container. Large modpacks like RLcraft will add significant resource pressure too.

**Comment by boldt** [comment]
I had the same issue (with another mod). In my case, the docker image was some month old. Thus, I ran:

```bash
docker pull itzg/minecraft-server
docker-compose up
```

Maybe it helps.

**Comment by jettero** [comment]
personally... I found it easiest (after trying that GENERIC modpack stuff) to unpack the RLcraft+Server zip, and drop the forge installer jar in a repack of the zip.

**Comment by howroyd** [comment]
> personally... I found it easiest (after trying that GENERIC modpack stuff) to unpack the RLcraft+Server zip, and drop the forge installer jar in a repack of the zip.

I tried that to no avail.  Having the same issue with a curseforge modpack:
```
howroyd@7gr-server:~$ ./docker-mc-better.sh && docker logs -f mc-better-fabric                                       71607053c386e5e0dac6fe2bba4a258daf4aa232766544d7f66bc4f79cea0cb0
[init] Running as uid=1000 gid=1000 with /data as 'drwxrwxr-x    6 1000     1000          4096 Jun 27 11:01 /data'
[init] Resolved version given 1.16.5 into 1.16.5
[init] Resolving type given CURSEFORGE
[init] Looking for Feed-The-Beast / CurseForge server modpack.
[init] Unpacking FTB server modpack /modpacks/Better+Minecraft+Server+Pack+[FABRIC]+v8.zip ...
................................................................................................................................................................................................................................................................................................................................................................................................................................
[init] Please make sure you are using the server version of the FTB modpack!
```

With the script:
```
#!/bin/sh

docker run -e EULA=TRUE -d \
  -v /usr/games/mc_better_fabric/data:/data \
  -v /usr/games/mc_better_fabric/modpacks:/modpacks \
  -e VERSION=1.16.5 \
  -e TYPE=CURSEFORGE \
  -e USE_MODPACK_START_SCRIPT=true \
  -e CF_SERVER_MOD=/modpacks/Better+Minecraft+Server+Pack+\[FABRIC\]+v8.zip \
  -e MEMORY=4G \
  -e WHITELIST=RyanL000 \
  -e ENFORCE_WHITELIST=true \
  -e OPS=GreenaGiant,CrazyRedJezz \
  -e MAX_PLAYERS=10 \
  -e PVP=FALSE \
  -e ALLOW_FLIGHT=TRUE \
  -e SERVER_NAME="Simon's Modded Server" \
  -e SERVER_PORT=25566 \
  -e RCON_PORT=25576 \
  --restart always -it \
  -p 25566:25566 --dns 8.8.8.8 \
  --name mc-better-fabric itzg/minecraft-server:java8
```

And I took the `fabric-installer-0.7.4.jar` and dropped it into the root of the `.zip` alongside the mod folder.

**Comment by delauraen** [comment]
> 
> 
> > personally... I found it easiest (after trying that GENERIC modpack stuff) to unpack the RLcraft+Server zip, and drop the forge installer jar in a repack of the zip.
> 
> I tried that to no avail. Having the same issue with a curseforge modpack:
> 
> ```
> howroyd@7gr-server:~$ ./docker-mc-better.sh && docker logs -f mc-better-fabric                                       71607053c386e5e0dac6fe2bba4a258daf4aa232766544d7f66bc4f79cea0cb0
> [init] Running as uid=1000 gid=1000 with /data as 'drwxrwxr-x    6 1000     1000          4096 Jun 27 11:01 /data'
> [init] Resolved version given 1.16.5 into 1.16.5
> [init] Resolving type given CURSEFORGE
> [init] Looking for Feed-The-Beast / CurseForge server modpack.
> [init] Unpacking FTB server modpack /modpacks/Better+Minecraft+Server+Pack+[FABRIC]+v8.zip ...
> ................................................................................................................................................................................................................................................................................................................................................................................................................................
> [init] Please make sure you are using the server version of the FTB modpack!
> ```
> 
> With the script:
> 
> ```
> #!/bin/sh
> 
> docker run -e EULA=TRUE -d \
>   -v /usr/games/mc_better_fabric/data:/data \
>   -v /usr/games/mc_better_fabric/modpacks:/modpacks \
>   -e VERSION=1.16.5 \
>   -e TYPE=CURSEFORGE \
>   -e USE_MODPACK_START_SCRIPT=true \
>   -e CF_SERVER_MOD=/modpacks/Better+Minecraft+Server+Pack+\[FABRIC\]+v8.zip \
>   -e MEMORY=4G \
>   -e WHITELIST=RyanL000 \
>   -e ENFORCE_WHITELIST=true \
>   -e OPS=GreenaGiant,CrazyRedJezz \
>   -e MAX_PLAYERS=10 \
>   -e PVP=FALSE \
>   -e ALLOW_FLIGHT=TRUE \
>   -e SERVER_NAME="Simon's Modded Server" \
>   -e SERVER_PORT=25566 \
>   -e RCON_PORT=25576 \
>   --restart always -it \
>   -p 25566:25566 --dns 8.8.8.8 \
>   --name mc-better-fabric itzg/minecraft-server:java8
> ```
> 
> And I took the `fabric-installer-0.7.4.jar` and dropped it into the root of the `.zip` alongside the mod folder.


I had this same issue, but I set the `USE_MODPACK_START_SCRIPT` to false and it was able to launch properly. I think this was mentioned in the readme that the startup scripts can be buggy.
